src/main/java/net/javadiscord/javabot/systems/commands/FormatCodeCommand.java

```diff
@@ -19,25 +19,38 @@
  * Command that allows members to format messages.
  */
 public class FormatCodeCommand implements ISlashCommand, IMessageContextCommand {
+	long id;
+
 	@Override
 	public ReplyCallbackAction handleSlashCommandInteraction(SlashCommandInteractionEvent event) {
 		var idOption = event.getOption("message-id");
 		var formatOption = event.getOption("format");
+		var slashConfig = Bot.config.get(event.getGuild()).getSlashCommand();
 		String format = formatOption == null ? "java" : formatOption.getAsString();
-		long id;
 		if (idOption == null) {
 			if (event.getChannel().hasLatestMessage()) {
-				id = event.getChannel().getLatestMessageIdLong();
+				event.getChannel().getHistory().retrievePast(100) // max limit
+						.map(messages -> messages)
+						.queue(messages -> {
+							for (Message message : messages) {
+								if (!message.getAuthor().isBot()){
+									id = message.getIdLong();
+									break;
+								}
+							}
+							event.getTextChannel().retrieveMessageById(id).queue(
```

Comment:
`id` may be `null` if all the messages were sent by the bot (could happen in channels like the suggestion channel or looking-for-programmer if #216 is implemented and deployed)

Reply:
alright imma add a null check

Reply:
Yeah, you're right

Reply:
it is handled actually
`e -> Responses.error(event.getHook(), "Could not retrieve message.").queue())`